_stbt/logging.py

```diff
@@ -69,6 +75,46 @@ def __call__(self, parser, namespace, values, option_string=None):
              "applied")
 
 
+class ImageLogger(object):
+    """Log intermediate images used in image processing (such as `match`).
+
+    Create a new ImageLogger instance for each frame of video.
+    """
+    _frame_number = 1
+
+    def __init__(self, name):
+        self.name = name
+        self.images = OrderedDict()
+        self.frame_number = ImageLogger._frame_number
+        ImageLogger._frame_number += 1
+
+    def add(self, name, image):
+        if get_debug_level() <= 1:
+            return
+        if name in self.images:
+            raise ValueError("Image for name '%s' already logged" % name)
+        with numpy_from_sample(image, readonly=True) as img:
+            self.images[name] = img.copy()
```

Comment:
Because it will allow me to pull additional debugging out of `_match` and into `MatchImageLogger`. Specifically this block of code:

    if logging.get_debug_level() > 1:
        source_with_roi = image.copy()
        _draw_match(source_with_roi, region, first_pass_matched, thickness=1)
        imglog.add("source_with_roi", source_with_roi)

I had forgotten to do that, so I'll push a fixup shortly.